wrangler-core/src/main/java/io/cdap/functions/DateAndTime.java

```diff
@@ -0,0 +1,663 @@
+   *
+   * @param source a {@link LocalDateTime} type
+   * @param destination a {@link LocalDateTime} type
+   * @return number of seconds between two timestamps.
+   */
+  public static Long SecondsSinceFromTimestamp(LocalDateTime source, LocalDateTime destination) {
+    return source == null || destination == null ? null : Duration.between(destination, source).getSeconds();
+  }
+
+  /**
+   * @return Returns the system time and date as a formatted string.
+   */
+  public static String TimeDate() {
+    return LocalDateTime.now().format(OUTPUT_DATETIME_FORMAT);
+  }
+
+  /**
+   * Returns a time from the given hours, minutes, seconds, and microseconds, given as four separate values.
+   * If any input is null, null will be returned.
+   *
+   * @param hours component of {@link LocalTime}
+   * @param mins component of {@link LocalTime}
+   * @param seconds component of {@link LocalTime}
+   * @param microseconds component of {@link LocalTime}
+   * @return a instance of {@link LocalTime}
+   */
+  public static LocalTime TimeFromComponents(Integer hours, Integer mins, Integer seconds, Integer microseconds) {
+    return hours == null || mins == null || seconds == null || microseconds == null ? null :
+             LocalTime.of(hours, mins, seconds, microseconds * 1000);
+  }
+
+  /**
+   * Returns the time given the number of seconds since midnight. If seconds is null, midnight will be returned.
+   *
+   * @param seconds since midnight.
+   * @return time since midnight.
+   */
+  public static LocalTime TimeFromMidnightSeconds(Integer seconds) {
+    return LocalTime.of(0,0,0).plusSeconds(seconds == null ? 0 : seconds);
+  }
+
+  /**
+   * Returns the time, with offsets applied from the base time.
+   * hour offset, minute offset, and second offset each given as separate values.
+   * The seconds offset can include partial seconds.
+   * If time is null, null will be returned. If the offset is null, it will be treated as 0.
+   *
+   * @param time baseline {@link LocalTime}
+   * @param hour offset to added to baseline.
+   * @param min offset to added to baseline.
+   * @param seconds offset to added to baseline.
+   * @return hour, min and second added to baseline.
+   */
+  public static LocalTime TimeOffsetByComponents(LocalTime time, Integer hour, Integer min, Integer seconds) {
+    return time == null ? null : time.plusHours(hour == null ? 0 : hour)
+                                   .plusMinutes(min == null ? 0 : min)
+                                   .plusSeconds(seconds == null ? 0 : seconds);
+  }
+
+  /**
+   * Returns a timestamp from the given date and time. If any input is null, null will be returned.
+   *
+   * @param date a instance of {@link LocalDate}
+   * @param time a instance of {@link LocalTime}
+   * @return a instance of {@link LocalDateTime}
+   */
+  public static LocalDateTime TimestampFromDateTime(LocalDate date, LocalTime time) {
```

Comment:
A timestamp normally represents an actual point in time, which different than what these functions do.

Reply:
The DataStage date and time functions basically do not operate on a ZonedDateTime, here is their doc: https://www.ibm.com/support/knowledgecenter/SSZJPZ_11.7.0/com.ibm.swg.im.iis.ds.parjob.dev.doc/topics/r_deeref_Date_and_Time_Functions.html.
We have also heard about use cases about these functions used in LocalDateTime. I feel these functions are more like the Datetime type in Bigquery.

Reply:
let's rename to `GetDateTime(LocalDate date, LocalTime time)`